doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -11,92 +11,83 @@ These tactics use the equality :g:`eq:forall A:Type, A->A->Prop` defined in
 file ``Logic.v`` (see :ref:`coq-library-logic`). The notation for :g:`eq T t u` is
 simply :g:`t=u` dropping the implicit type of :g:`t` and :g:`u`.
 
-.. tacn:: rewrite @term
-   :name: rewrite
+.. tacn:: rewrite {+, @oriented_rewriter } {? @occurrences } {? by @ltac_expr3 }
 
-   This tactic applies to any goal. The type of :token:`term` must have the form
+   .. insertprodn oriented_rewriter constr_with_bindings_arg
 
-   ``forall (x``:sub:`1` ``:A``:sub:`1` ``) ... (x``:sub:`n` ``:A``:sub:`n` ``), eq term``:sub:`1` ``term``:sub:`2` ``.``
+   .. prodn::
+      oriented_rewriter ::= {? {| -> | <- } } {? @natural } {? {| ? | ! } } @constr_with_bindings_arg
+      constr_with_bindings_arg ::= {? > } @one_term {? with @bindings }
```

Comment:
I suggest calling this `term_with_bindings`. I would also like to understand what the `>` does in this case.

Reply:
How about `one_term_with_bindings`?

Reply:
Sure.